{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "34ad3351-9913-42ee-a394-111d61cff3fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,sys\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_rows',10) \n",
    "pd.set_option('display.max_columns',20) \n",
    "pd.set_option('display.width', None) \n",
    "pd.set_option('display.max_colwidth',None)\n",
    "import matplotlib.pylab as plt\n",
    "import glob\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import yaml\n",
    "mpl.style.use('default')\n",
    "mpl.rcParams['pdf.fonttype'] = 42\n",
    "mpl.rcParams['ps.fonttype'] = 42\n",
    "mpl.rcParams['font.family'] = 'sans-serif'\n",
    "mpl.rcParams['font.sans-serif'] = 'Arial'\n",
    "mpl.rcParams['figure.dpi'] = 80\n",
    "mpl.rcParams['savefig.dpi']=300"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da13f45b-936a-4e4a-bf4f-e130e281b62d",
   "metadata": {},
   "source": [
    "Path\n",
    "```shell\n",
    "# meQTL (20240713): Permutation\n",
    "Trans: /anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/*.SNP_DMRfileter.tsv\n",
    "Cis: /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/*.filtered.tsv\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3939b90a-f1fc-49ba-8003-8e26e0edfe67",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## 1. Download GWAS statistic summary from GWAS Catalog and Processing\n",
    "https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/harmonised_list.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5a5cad3b-c442-478c-ab7f-880cae59f745",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df=pd.read_csv(\"https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/harmonised_list.txt\",sep='\\t',header=None,names=['url'])\n",
    "# df['real_url']=\"https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/\"+df.url.apply(lambda x:x.lstrip('./')).map(str)\n",
    "# df.real_url.to_csv(\"gwas_catalog_harmonised_urls.tsv\",header=False,sep='\\t',index=False)\n",
    "# df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f20bbc4d-e080-4613-aa04-3cdd9a634b47",
   "metadata": {},
   "source": [
    "```shell\n",
    "# downloading\n",
    "rm -f cmd\n",
    "while read url; do\n",
    "    # echo ${url}\n",
    "    echo \"wget -nc -P gwas ${url}\" >> cmd\n",
    "    echo \"wget -nc -P gwas ${url}-meta.yaml\" >> cmd\n",
    "done < gwas_catalog_harmonised_urls.tsv\n",
    "\n",
    "cat cmd | parallel -j 24\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e274d281-c145-4922-8fa8-df87603cf93d",
   "metadata": {},
   "source": [
    "color input format:\n",
    "- category:\n",
    "\n",
    "(1): snp,position,MAF,pvalues,N,type='cc', and s(for a case control dataset, the proportion of samples in dataset 1 that are cases) </br1>\n",
    "\n",
    "test_data1=list(snp='snp1',position=100,MAF=0.998,pvalues=4e-8,N=8742,type='cc',s=26948/(26948+8724)) #beta=2.16 </br1>\n",
    "print(check_dataset(test_data1)) \n",
    "\n",
    "(2): snp,position,beta,varbeta </br1>\n",
    "\n",
    "test_data1=list(snp='snp1',position=100,beta=2.16,varbeta=0.1,type='cc') #MAF=0.998,N=8742,pvalues=4e-8,\n",
    "Preferably, give beta and varbeta\n",
    "print(check_dataset(test_data1))\n",
    "\n",
    "\n",
    "dataset1=list(beta=effect_size, varbeta=standard_error^2,type=\"cc\",MAF=frequency,N=sample_size,s=cases/sample_size\n",
    "\n",
    "- Quant\n",
    "\n",
    "(1): test_data1=list(snp='snp1',position=100,beta=2.16,varbeta=0.1,type='quant',MAF=0.998,N=8742)\n",
    "\n",
    "(2): test_data1=list(snp='snp1',position=100,type='quant',MAF=0.998,N=8742,pvalues=4e-8)\n",
    "\n",
    "N: Number of samples in dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9af337a8-9874-4ca4-9f1f-b2d7d57c662b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>gwas_id</th>\n",
       "      <th>trait_description</th>\n",
       "      <th>ontology_mapping</th>\n",
       "      <th>genome_assembly</th>\n",
       "      <th>sample_size</th>\n",
       "      <th>is_harmonised</th>\n",
       "      <th>case_control_study</th>\n",
       "      <th>n_case</th>\n",
       "      <th>n_control</th>\n",
       "      <th>case_count</th>\n",
       "      <th>control_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>34662886-GCST90080617-EFO_0000284.h.tsv.gz</td>\n",
       "      <td>GCST90080617</td>\n",
       "      <td>['ICD10 N40: Benign prostatic hyperplasia']</td>\n",
       "      <td>['EFO_0000284']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>172320.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15936.0</td>\n",
       "      <td>156384.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>34662886-GCST90080307-EFO_0000408.h.tsv.gz</td>\n",
       "      <td>GCST90080307</td>\n",
       "      <td>['ICD10 K83.8: Other specified diseases of biliary tract']</td>\n",
       "      <td>['EFO_0009534']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>387930.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>386668.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCST90265493.h.tsv.gz</td>\n",
       "      <td>GCST90265493</td>\n",
       "      <td>['Urine methionine sulfone levels in chronic kidney disease']</td>\n",
       "      <td>['EFO_0800072']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4888.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCST90266242.h.tsv.gz</td>\n",
       "      <td>GCST90266242</td>\n",
       "      <td>['Plasma X-12680 levels in chronic kidney disease']</td>\n",
       "      <td>['EFO_0021320']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4934.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>35078996-GCST90089261-EFO_0007937.h.tsv.gz</td>\n",
       "      <td>GCST90089261</td>\n",
       "      <td>['Serum levels of protein RLN3']</td>\n",
       "      <td>['EFO_0007937']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>5349.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29396</th>\n",
       "      <td>34503513-GCST90060952-EFO_0004530.h.tsv.gz</td>\n",
       "      <td>GCST90060952</td>\n",
       "      <td>['Triacylglycerol(53:3)_[M+NH4]1+ levels']</td>\n",
       "      <td>['EFO_0004530']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>5662.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29397</th>\n",
       "      <td>32929287-GCST90001566-EFO_0007937.h.tsv.gz</td>\n",
       "      <td>GCST90001566</td>\n",
       "      <td>['Naive CD4-CD8- T cell Absolute Count']</td>\n",
       "      <td>['EFO_0004587']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>3395.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29398</th>\n",
       "      <td>33303764-GCST90010113-EFO_0007937.h.tsv.gz</td>\n",
       "      <td>GCST90010113</td>\n",
       "      <td>['CD40 ligand']</td>\n",
       "      <td>['EFO_0007937']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>1301.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29399</th>\n",
       "      <td>33414548-GCST90010726-EFO_0009777.h.tsv.gz</td>\n",
       "      <td>GCST90010726</td>\n",
       "      <td>['Citrulline levels']</td>\n",
       "      <td>['EFO_0009777']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>30940.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29400</th>\n",
       "      <td>GCST90265471.h.tsv.gz</td>\n",
       "      <td>GCST90265471</td>\n",
       "      <td>['Plasma malonylcarnitine levels in chronic kidney disease']</td>\n",
       "      <td>['EFO_0800232']</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4869.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29401 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         filename       gwas_id  \\\n",
       "0      34662886-GCST90080617-EFO_0000284.h.tsv.gz  GCST90080617   \n",
       "1      34662886-GCST90080307-EFO_0000408.h.tsv.gz  GCST90080307   \n",
       "2                           GCST90265493.h.tsv.gz  GCST90265493   \n",
       "3                           GCST90266242.h.tsv.gz  GCST90266242   \n",
       "4      35078996-GCST90089261-EFO_0007937.h.tsv.gz  GCST90089261   \n",
       "...                                           ...           ...   \n",
       "29396  34503513-GCST90060952-EFO_0004530.h.tsv.gz  GCST90060952   \n",
       "29397  32929287-GCST90001566-EFO_0007937.h.tsv.gz  GCST90001566   \n",
       "29398  33303764-GCST90010113-EFO_0007937.h.tsv.gz  GCST90010113   \n",
       "29399  33414548-GCST90010726-EFO_0009777.h.tsv.gz  GCST90010726   \n",
       "29400                       GCST90265471.h.tsv.gz  GCST90265471   \n",
       "\n",
       "                                                   trait_description  \\\n",
       "0                        ['ICD10 N40: Benign prostatic hyperplasia']   \n",
       "1         ['ICD10 K83.8: Other specified diseases of biliary tract']   \n",
       "2      ['Urine methionine sulfone levels in chronic kidney disease']   \n",
       "3                ['Plasma X-12680 levels in chronic kidney disease']   \n",
       "4                                   ['Serum levels of protein RLN3']   \n",
       "...                                                              ...   \n",
       "29396                     ['Triacylglycerol(53:3)_[M+NH4]1+ levels']   \n",
       "29397                       ['Naive CD4-CD8- T cell Absolute Count']   \n",
       "29398                                                ['CD40 ligand']   \n",
       "29399                                          ['Citrulline levels']   \n",
       "29400   ['Plasma malonylcarnitine levels in chronic kidney disease']   \n",
       "\n",
       "      ontology_mapping genome_assembly  sample_size  is_harmonised  \\\n",
       "0      ['EFO_0000284']          GRCh38     172320.0           True   \n",
       "1      ['EFO_0009534']          GRCh38     387930.0           True   \n",
       "2      ['EFO_0800072']          GRCh38       4888.0           True   \n",
       "3      ['EFO_0021320']          GRCh38       4934.0           True   \n",
       "4      ['EFO_0007937']          GRCh38       5349.0           True   \n",
       "...                ...             ...          ...            ...   \n",
       "29396  ['EFO_0004530']          GRCh38       5662.0           True   \n",
       "29397  ['EFO_0004587']          GRCh38       3395.0           True   \n",
       "29398  ['EFO_0007937']          GRCh38       1301.0           True   \n",
       "29399  ['EFO_0009777']          GRCh38      30940.0           True   \n",
       "29400  ['EFO_0800232']          GRCh38       4869.0           True   \n",
       "\n",
       "      case_control_study   n_case  n_control  case_count  control_count  \n",
       "0                    NaN  15936.0   156384.0         NaN            NaN  \n",
       "1                    NaN   1262.0   386668.0         NaN            NaN  \n",
       "2                  False      NaN        NaN         NaN            NaN  \n",
       "3                  False      NaN        NaN         NaN            NaN  \n",
       "4                    NaN      NaN        NaN         NaN            NaN  \n",
       "...                  ...      ...        ...         ...            ...  \n",
       "29396                NaN      NaN        NaN         NaN            NaN  \n",
       "29397                NaN      NaN        NaN         NaN            NaN  \n",
       "29398                NaN      NaN        NaN         NaN            NaN  \n",
       "29399                NaN      NaN        NaN         NaN            NaN  \n",
       "29400              False      NaN        NaN         NaN            NaN  \n",
       "\n",
       "[29401 rows x 12 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load meta\n",
    "import requests\n",
    "indir=\"gwas\"\n",
    "if not os.path.exists(\"gwas_catalog_harmonised_metadata.tsv\"):\n",
    "    R=[]\n",
    "    for file in glob.glob(f\"{indir}/*-meta.yaml\"):\n",
    "        sname=os.path.basename(file.rstrip('-meta.yaml'))\n",
    "        # print(file)\n",
    "        try:\n",
    "            with open(file) as f:\n",
    "                # params = yaml.load(f, Loader=yaml.FullLoader)\n",
    "                params = yaml.safe_load(f)\n",
    "        except:\n",
    "            continue\n",
    "        try:\n",
    "            sample_info=params['samples'][0]\n",
    "        except:\n",
    "            sample_info={}\n",
    "        url=params.get('gwas_catalog_api',None)\n",
    "        if not url is None:\n",
    "            try:\n",
    "                r=requests.get(url)\n",
    "                initialSampleSize=r.json()['initialSampleSize']\n",
    "                case,control=initialSampleSize.split(', ')\n",
    "                if 'cases' in case.lower() and 'controls' in control.lower():\n",
    "                    n_case=int(case.split(' ')[0].replace(',',''))\n",
    "                    n_control=int(control.split(' ')[0].replace(',',''))\n",
    "            except:\n",
    "                n_case=None;n_control=None\n",
    "        else:\n",
    "            n_case=None;n_control=None\n",
    "        R.append([sname,params['gwas_id'],params.get('trait_description',''),params.get('ontology_mapping',''),params.get('genome_assembly',''),\n",
    "                 sample_info.get('sample_size',np.nan),params.get('is_harmonised',np.nan),sample_info.get('case_control_study',np.nan),\n",
    "                 n_case,n_control,sample_info.get('case_count',np.nan),sample_info.get('control_count',np.nan)])\n",
    "    meta=pd.DataFrame(R,columns=['filename','gwas_id','trait_description','ontology_mapping','genome_assembly','sample_size','is_harmonised',\n",
    "                                 'case_control_study','n_case','n_control','case_count','control_count'])\n",
    "    meta.to_csv(\"gwas_catalog_harmonised_metadata.tsv\",sep='\\t',index=False)\n",
    "else:\n",
    "    meta=pd.read_csv(\"gwas_catalog_harmonised_metadata.tsv\",sep='\\t')\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c185f20e-5757-4458-86bb-0fe62643873c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NaN      26489\n",
       "False     2657\n",
       "True       255\n",
       "Name: case_control_study, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.case_control_study.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d245036f-59a1-4908-9a5c-fb8f1b9af05a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1399003/3299161378.py:9: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join('gwas',file),sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "meta=pd.read_csv(\"gwas_catalog_harmonised_metadata.tsv\",sep='\\t')\n",
    "hm_usecols=['hm_rsid','hm_variant_id','hm_chrom','hm_pos','hm_other_allele','hm_effect_allele','hm_effect_allele_frequency','hm_odds_ratio','hm_beta',]\n",
    "usecols=['rsid','variant_id','chromosome','base_pair_location','other_allele','effect_allele','effect_allele_frequency','odds_ratio','beta']\n",
    "cols=['rsid','variant_id','chrom','position','other_allele','effect_allele','MAF','odds_ratio','beta']\n",
    "common_cols=['p_value','standard_error']\n",
    "outdir=\"/anvil/projects/x-mcb130189/Wubin/gwas\"\n",
    "for category in ['cc','quant']:\n",
    "    if category=='cc':\n",
    "        # category: snp='snp1',position=100,MAF=0.998,pvalues=4e-8,N=8742,type='cc',s=26948/(26948+8724)\n",
    "        # or snp,position,beta,varbeta\n",
    "        meta1=meta.loc[(meta.n_case.notna()) & (meta.n_control.notna())]\n",
    "    else:\n",
    "        # Quant:snp='snp1',position=100,beta=2.16,varbeta=0.1,type='quant',MAF=0.998,N=8742\n",
    "        # or: snp='snp1',position=100,type='quant',MAF=0.998,N=8742,pvalues=4e-8\n",
    "        meta1=meta.loc[(meta.n_case.isna()) & (meta.n_control.isna())]\n",
    "    for gwas_id,file in meta1.loc[:,['gwas_id','filename']].values:\n",
    "        if not os.path.exists(os.path.join('gwas',file)):\n",
    "            continue\n",
    "        if os.path.exists(os.path.join(outdir,f\"gwas_catalog_{category}_no_beta\",gwas_id)) or os.path.exists(\n",
    "            os.path.join(outdir,f\"gwas_catalog_{category}_with_beta\",gwas_id)):\n",
    "            os.remove(os.path.join('gwas',file))\n",
    "            continue\n",
    "        try:\n",
    "            df=pd.read_csv(os.path.join('gwas',file),sep='\\t')\n",
    "            n1=len([col for col in df.columns if col in hm_usecols])\n",
    "            n2=len([col for col in df.columns if col in usecols])\n",
    "            if n2 > n1:\n",
    "                df=df.reindex(columns=usecols+common_cols)\n",
    "            else:\n",
    "                df=df.reindex(columns=hm_usecols+common_cols)\n",
    "            df.columns=cols+common_cols\n",
    "            df['FDR']=df.p_value.astype(float) * df.shape[0] # bonferroni\n",
    "            df=df.loc[df.FDR <= 0.05]\n",
    "            if df.shape[0]==0:\n",
    "                os.remove(os.path.join('gwas',file))\n",
    "                continue\n",
    "            # df.insert(0,'gwas_id',gwas_id)\n",
    "            beta_na_sum=df.beta.notna().sum()\n",
    "            if beta_na_sum == 0:\n",
    "                outdir1=os.path.join(outdir,f\"gwas_catalog_{category}_no_beta\")\n",
    "                df.drop('beta',axis=1,inplace=True)\n",
    "                df=df.loc[df.MAF.notna()]\n",
    "                df=df.loc[df.p_value.notna()]\n",
    "            else:\n",
    "                outdir1=os.path.join(outdir,f\"gwas_catalog_{category}_with_beta\")\n",
    "            if not os.path.exists(outdir1):\n",
    "                os.mkdir(outdir1)\n",
    "            df=df.loc[df.rsid.notna()]\n",
    "            if df.shape[0]==0:\n",
    "                os.remove(os.path.join('gwas',file))\n",
    "                continue\n",
    "            df.to_csv(os.path.join(outdir1,gwas_id),sep='\\t',index=False)\n",
    "            os.remove(os.path.join('gwas',file))\n",
    "        except:\n",
    "            print(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e7c52ffb-2966-440d-9945-4bfa676b9b67",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gwas_catalog_cc_no_beta\n",
      "794\n",
      "gwas_catalog_cc_with_beta\n",
      "817\n",
      "gwas_catalog_quant_no_beta\n",
      "17\n",
      "gwas_catalog_quant_with_beta\n",
      "9358\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
      "/tmp/ipykernel_2393043/815725759.py:13: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "# merge results\n",
    "def parse_chrom(x):\n",
    "    try:\n",
    "        return \"chr\"+str(int(x))\n",
    "    except:\n",
    "        return \"chr\"+str(x)\n",
    "for category in ['cc','quant']:\n",
    "    for folder_name in [f\"gwas_catalog_{category}_no_beta\",f\"gwas_catalog_{category}_with_beta\"]:\n",
    "        print(folder_name)\n",
    "        R=[]\n",
    "        print(len(os.listdir(os.path.join(\"gwas\",folder_name))))\n",
    "        for file in os.listdir(os.path.join(\"gwas\",folder_name)):\n",
    "            df=pd.read_csv(os.path.join(\"gwas\",folder_name,file),sep='\\t')\n",
    "            # print(df.shape[0])\n",
    "            df.insert(0,'gwas_id',file)\n",
    "            df.chrom=df.chrom.apply(parse_chrom)\n",
    "            df.position=df.position.map(int)\n",
    "            df.rename(columns={'rsid':'snp','p_value':'pvalues'},inplace=True)\n",
    "            if folder_name.endswith('with_beta'):\n",
    "                df['varbeta']=df.standard_error.apply(np.sqrt)\n",
    "            R.append(df)\n",
    "        df=pd.concat(R)\n",
    "        df.to_csv(os.path.join(\"gwas_catalog\",f\"{folder_name}.tsv.gz\"),sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcd3241d-fe2b-4715-a9ed-3c8e17788ffe",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## 2. Read MAF & snp positions from vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "bac0222f-9dfe-477b-9d14-693cdd65fa2c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/02.Genotypes/all_auto.vcf.gz\n"
     ]
    }
   ],
   "source": [
    "! ls /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/02.Genotypes/all_auto.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "e535ec32-2010-44f4-a33c-680fa75a9bec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Chrom</th>\n",
       "      <th>POS</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs147880041</td>\n",
       "      <td>chr10</td>\n",
       "      <td>47317</td>\n",
       "      <td>0.3840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs9329305</td>\n",
       "      <td>chr10</td>\n",
       "      <td>47876</td>\n",
       "      <td>0.4995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs4431953</td>\n",
       "      <td>chr10</td>\n",
       "      <td>48005</td>\n",
       "      <td>0.1670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs145359804</td>\n",
       "      <td>chr10</td>\n",
       "      <td>48219</td>\n",
       "      <td>0.4995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10904045</td>\n",
       "      <td>chr10</td>\n",
       "      <td>48486</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID  Chrom    POS     MAF\n",
       "0  rs147880041  chr10  47317  0.3840\n",
       "1    rs9329305  chr10  47876  0.4995\n",
       "2    rs4431953  chr10  48005  0.1670\n",
       "3  rs145359804  chr10  48219  0.4995\n",
       "4   rs10904045  chr10  48486  0.0000"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv(\"/anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/02.Genotypes/all_auto.vcf.gz\",sep='\\t',comment=\"#\",usecols=[0,1,2,7])\n",
    "df.columns=['chrom','pos','rsid','info']\n",
    "df['varid']=df.chrom.map(str)+'-'+df.pos.map(str)\n",
    "df['MAF']=df['info'].apply(lambda x:float(x.split('MAF=')[1].split(';')[0]))\n",
    "df=df.loc[:,['chrom','pos','varid','rsid','MAF']].drop_duplicates()\n",
    "df.sort_values(['chrom','pos'],ascending=True,inplace=True)\n",
    "df.to_csv(\"MAF.tsv\",sep='\\t',index=False)\n",
    "# sort \n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ae95755-13c0-4ce8-b3a7-a2f4f32d28a4",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 3. Preprocessing for cis-meQTLs & trans-meQTl (permutation)\n",
    "Cis: /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/*.filtered.tsv\n",
    "\n",
    "Trans: /anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/*.SNP_DMRfileter.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4282fabf-c065-4b0e-b35b-35d3dbdc716d",
   "metadata": {},
   "source": [
    "- Quant\n",
    "\n",
    "(1): test_data1=list(snp='snp1',position=100,beta=2.16,varbeta=0.1,type='quant',MAF=0.998,N=8742)\n",
    "\n",
    "(2): test_data1=list(snp='snp1',position=100,type='quant',MAF=0.998,N=8742,pvalues=4e-8)\n",
    "\n",
    "N: Number of samples in dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5072a30f-c212-4989-b197-cb496dd2ab0e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.5M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/B-Mem.permutation.filtered.tsv\n",
      "2.5M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/B-Naive.permutation.filtered.tsv\n",
      "2.0M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/NK-cell1.permutation.filtered.tsv\n",
      "2.5M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/NK-cell2.permutation.filtered.tsv\n",
      "9.0M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/Tc-Mem.permutation.filtered.tsv\n",
      "4.0M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/Tc-Naive.permutation.filtered.tsv\n",
      "3.5M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/Th-Mem.permutation.filtered.tsv\n",
      "4.0M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/Th-Naive.permutation.filtered.tsv\n"
     ]
    }
   ],
   "source": [
    "! ls /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/*.filtered.tsv -sh\n",
    "# N = 110"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d9d7c302-56ca-40c8-afa4-e6f26ff959c6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phe_id\tphe_chr\tphe_from\tphe_to\tphe_strd\tn_var_in_cis\tdist_phe_var\tvar_id\tvar_chr\tvar_from\tvar_to\tdof1\tdof2\tbml1\tbml2\tnom_pval\tr_squared\tslope\tadj_emp_pval\tadj_beta_pval\n",
      "dmr492247\tchr4\t171137902\t171137901\t+\t15931\t-765908.0\trs10579886\tchr4\t170371992.0\t170371994.0\t105.0\t119.581\t0.965135\t3577.44\t3.01516e-05\t0.153439\t0.407429\t0.0329967\t0.0340715\n",
      "dmr492339\tchr4\t171850035\t171850034\t+\t15154\t99940.0\trs333003\tchr4\t171949974.0\t171949974.0\t105.0\t119.396\t0.927341\t5477.14\t6.64507e-06\t0.176484\t0.361367\t0.0125987\t0.012289\n",
      "dmr492484\tchr4\t172714094\t172714093\t+\t13965\t-198227.0\trs28726202\tchr4\t172515867.0\t172515867.0\t105.0\t118.597\t0.942498\t6391.17\t9.23566e-06\t0.171509\t-0.353729\t0.0214979\t0.0200405\n",
      "dmr492511\tchr4\t172802092\t172802091\t+\t14085\t-413115.0\trs17058366\tchr4\t172388977.0\t172388977.0\t105.0\t95.5972\t0.898196\t3303.98\t3.05512e-06\t0.188131\t0.705038\t0.0484952\t0.0414824\n",
      "dmr492561\tchr4\t173019778\t173019777\t+\t13910\t662344.0\trs73004456\tchr4\t173682121.0\t173682121.0\t105.0\t113.439\t0.901351\t7196.52\t5.20351e-06\t0.180164\t2.28574\t0.029997\t0.0242944\n",
      "dmr492597\tchr4\t173152474\t173152473\t+\t14040\t-814675.0\trs72702892\tchr4\t172337799.0\t172337799.0\t105.0\t46.1528\t1.16238\t504.517\t1.64792e-09\t0.293763\t1315.71\t0.00619938\t0.0178406\n",
      "dmr492659\tchr4\t173516809\t173516808\t+\t13902\t177392.0\trs7437913\tchr4\t173694200.0\t173694200.0\t105.0\t111.778\t0.959182\t2092.79\t2.05114e-05\t0.159355\t0.48512\t0.0273973\t0.0271557\n",
      "dmr492665\tchr4\t173531231\t173531365\t+\t13913\t737956.0\trs35194028\tchr4\t174269321.0\t174269322.0\t105.0\t115.285\t0.988787\t4535.28\t2.62082e-05\t0.155595\t0.265128\t0.0516948\t0.0484534\n",
      "dmr492723\tchr4\t173780342\t173780341\t+\t14307\t284870.0\trs11930038\tchr4\t174065211.0\t174065211.0\t105.0\t114.505\t0.983207\t2605.02\t1.88159e-05\t0.160676\t-0.68238\t0.0241976\t0.0218017\n"
     ]
    }
   ],
   "source": [
    "! head /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/B-Mem.permutation.filtered.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1d6acf74-8724-4c10-bd83-a72254a8c84e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "outdir=os.path.abspath(\"cis-meQTL\")\n",
    "if not os.path.exists(outdir):\n",
    "    os.mkdir(outdir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7ab7e781-5dc2-4ebb-b114-a46dfb7874d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(\"MAF.tsv\",sep='\\t',index_col='rsid',dtype={'pos':int})\n",
    "maf_dict=df.MAF.to_dict()\n",
    "snp_positions=df.pos.to_dict()\n",
    "snp_chrom=df.chrom.to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "114f3a2f-8faa-4035-9067-571628726b59",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B-Mem\n",
      "Tc-Naive\n",
      "B-Naive\n",
      "Th-Naive\n",
      "Th-Mem\n",
      "Tc-Mem\n",
      "NK-cell1\n",
      "NK-cell2\n"
     ]
    }
   ],
   "source": [
    "for file in glob.glob(\"/anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/02.permute/*.filtered.tsv\"):\n",
    "    group=os.path.basename(file).split('.')[0]\n",
    "    print(group)\n",
    "    df=pd.read_csv(file,sep='\\t',usecols=['phe_id','var_id','nom_pval']) #'slope','slope_se'\n",
    "    df.rename(columns={'phe_id':'dmr_id','var_id':'snp','nom_pval':'pvalues'},inplace=True)\n",
    "    df['MAF']=df.snp.map(maf_dict)\n",
    "    df['chrom']=df.snp.map(snp_chrom)\n",
    "    df['position']=df.snp.map(snp_positions)\n",
    "    df.to_csv(os.path.join(outdir,f\"{group}.tsv\"),sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "71479feb-7ce3-4049-9e60-309a33fa58e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/B-Mem.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/B-Naive.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/Monocyte.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/NK-cell1.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/NK-cell2.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/Tc-Mem.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/Tc-Naive.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/Th-Mem.transQTL.FDRfilter.SNP_DMRfileter.tsv\n",
      "/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/Th-Naive.transQTL.FDRfilter.SNP_DMRfileter.tsv\n"
     ]
    }
   ],
   "source": [
    "! ls /anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/*.SNP_DMRfileter.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6fda551a-d47b-4e2c-9a16-8614410f1644",
   "metadata": {},
   "outputs": [],
   "source": [
    "outdir=os.path.abspath(\"trans-meQTL\")\n",
    "if not os.path.exists(outdir):\n",
    "    os.mkdir(outdir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2c6df6ab-fda3-443e-bdbd-e7dbd08dd096",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B-Naive\n",
      "Monocyte\n",
      "NK-cell2\n",
      "Tc-Naive\n",
      "NK-cell1\n",
      "Tc-Mem\n",
      "Th-Mem\n",
      "Th-Naive\n",
      "B-Mem\n"
     ]
    }
   ],
   "source": [
    "for file in glob.glob(\"/anvil/scratch/x-wangwl/02.mQTL/07.tensorqtl/*.SNP_DMRfileter.tsv\"):\n",
    "    group=os.path.basename(file).split('.')[0]\n",
    "    print(group)\n",
    "    df=pd.read_csv(file,sep='\\t',usecols=['phenotype_id','variant_id','pval']) #'slope','slope_se'\n",
    "    df.rename(columns={'phenotype_id':'dmr_id','variant_id':'snp','pval':'pvalues'},inplace=True)\n",
    "    df['MAF']=df.snp.map(maf_dict)\n",
    "    df['chrom']=df.snp.map(snp_chrom)\n",
    "    df['position']=df.snp.map(snp_positions)\n",
    "    df.to_csv(os.path.join(outdir,f\"{group}.tsv\"),sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb8cd055-dcc2-4c4b-89e4-5e980488cfc4",
   "metadata": {},
   "source": [
    "## 4. Preprocessing for cis-meQTLs & trans-meQTl (norminal)\n",
    "/anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/*.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "07eaf92a-3fff-4a5b-897c-99dbc865f8d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "127M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/B-Mem.fdr.csv\n",
      "112M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/B-Naive.fdr.csv\n",
      "4.3G /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/Monocyte.fdr.csv\n",
      "2.5M /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/NK-cell1.fdr.csv\n",
      "1.3G /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/NK-cell2.fdr.csv\n",
      " 47G /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/Tc-Mem.fdr.csv\n",
      " 25G /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/Tc-Naive.fdr.csv\n",
      " 22G /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/Th-Mem.fdr.csv\n",
      " 30G /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/Th-Naive.fdr.csv\n"
     ]
    }
   ],
   "source": [
    "! ls -sh /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/*.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "432b8194-b877-4394-a9c6-bdde83190eac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dmrID\tpchr\tpstart\tpend\tstrand\tnvar\tdist\tvid\tvchr\tvstart\tvend\tpval\tr2\tslope\tbest\tfdr\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t795692\t.\tchr1\t813144\t813144\t0.00839365\t0.0625816\t-3.06319\t0\t0.009094045820125712\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t819316\t.\tchr1\t836768\t836768\t0.00680565\t0.065855\t-1.49166\t0\t0.008079236811731378\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t826633\t.\tchr1\t844085\t844085\t0.00502703\t0.0705884\t-59.0295\t0\t0.006755826919850406\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t830354\t.\tchr1\t847806\t847806\t0.00917997\t0.0611852\t-0.599588\t0\t0.009550839424280826\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t839044\t.\tchr1\t856496\t856496\t0.00917997\t0.0611852\t-0.538202\t0\t0.009550839424280826\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t843344\t.\tchr1\t860796\t860796\t0.00912118\t0.0612854\t-0.538662\t0\t0.009517615571173045\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t848304\t.\tchr1\t865756\t865756\t9.41439e-06\t0.166865\t-0.417911\t0\t5.652347428016576e-05\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t853135\t.\tchr1\t870587\t870587\t0.00587675\t0.0681474\t-0.282233\t0\t0.007416763368714343\n",
      "dmr0\tchr1\t17453\t17452\t+\t3289\t853492\t.\tchr1\t870944\t870944\t0.00917997\t0.0611852\t-8.37203\t0\t0.009550839424280826\n"
     ]
    }
   ],
   "source": [
    "! head /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/Th-Naive.fdr.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "260e242a-3679-4a91-a74c-9ccbf587dc3c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "outdir=os.path.abspath(\"cis-meQTL-norminal\")\n",
    "if not os.path.exists(outdir):\n",
    "    os.mkdir(outdir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "73582784-e79a-4e2e-9cb4-c7288beb1508",
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(\"MAF.tsv\",sep='\\t',dtype={'POS':int})\n",
    "maf_dict=df.set_index('rsid').MAF.to_dict()\n",
    "vpos2snp=df.set_index(['chrom','pos']).ID.to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "for file in `ls /anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/*.csv`; do\n",
    "    bname=$(basename ${file})\n",
    "    group=${bname/.*}\n",
    "    echo ${group}\n",
    "    awk 'BEGIN{FS=OFS=\"\\t\"}; FILENAME==ARGV[1]{D1[$3]=$4; D2[$3]=$5; next} FILENAME==ARGV[2]{key=$1\"-\"$2; print($1,$2,$3,D1[key],D2[key],$4)}' ../MAF.tsv <(awk 'BEGIN {FS=OFS=\"\\t\"}; {print($9,$10,$1,$12)}' ${file} | sed '1d') | awk 'BEGIN {FS=OFS=\"\\t\"}; {if($4!=\"\") {print($0)}}' > ${group}.tsv\n",
    "done;\n",
    "\n",
    "# columns: ['chrom','position','dmr_id','snp','MAF','pvalues']"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "dea0e7ef0614b6c8"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c66a424-33ed-44d1-bed5-f4ac3390e5ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NK-cell1\n",
      "Tc-Naive\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f376c465460>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/anvil/projects/x-mcb130189/Wubin/Software/miniconda3/envs/m3c/lib/python3.9/site-packages/ipykernel/ipkernel.py\", line 775, in _clean_thread_parent_frames\n",
      "    def _clean_thread_parent_frames(\n",
      "KeyboardInterrupt: \n",
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f376c465460>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/anvil/projects/x-mcb130189/Wubin/Software/miniconda3/envs/m3c/lib/python3.9/site-packages/ipykernel/ipkernel.py\", line 775, in _clean_thread_parent_frames\n",
      "    def _clean_thread_parent_frames(\n",
      "KeyboardInterrupt: \n",
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f376c465460>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/anvil/projects/x-mcb130189/Wubin/Software/miniconda3/envs/m3c/lib/python3.9/site-packages/ipykernel/ipkernel.py\", line 775, in _clean_thread_parent_frames\n",
      "    def _clean_thread_parent_frames(\n",
      "KeyboardInterrupt: \n",
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f376c465460>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/anvil/projects/x-mcb130189/Wubin/Software/miniconda3/envs/m3c/lib/python3.9/site-packages/ipykernel/ipkernel.py\", line 775, in _clean_thread_parent_frames\n",
      "    def _clean_thread_parent_frames(\n",
      "KeyboardInterrupt: \n",
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f376c465460>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/anvil/projects/x-mcb130189/Wubin/Software/miniconda3/envs/m3c/lib/python3.9/site-packages/ipykernel/ipkernel.py\", line 775, in _clean_thread_parent_frames\n",
      "    def _clean_thread_parent_frames(\n",
      "KeyboardInterrupt: \n",
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f376c465460>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/anvil/projects/x-mcb130189/Wubin/Software/miniconda3/envs/m3c/lib/python3.9/site-packages/ipykernel/ipkernel.py\", line 775, in _clean_thread_parent_frames\n",
      "    def _clean_thread_parent_frames(\n",
      "KeyboardInterrupt: \n"
     ]
    }
   ],
   "source": [
    "# for file in glob.glob(\"/anvil/scratch/x-wangwl/02.mQTL/04.bulkQTL/04.cis/00.FDR/*.csv\"):\n",
    "#     group=os.path.basename(file).split('.')[0]\n",
    "#     print(group)\n",
    "#     outfile=os.path.join(outdir,f\"{group}.tsv\")\n",
    "#     if os.path.exists(outfile):\n",
    "#         os.remove(outfile)\n",
    "#     for df in pd.read_csv(file,sep='\\t',usecols=['dmrID','vchr','vstart','pval'],chunksize=50000): #'slope','slope_se'\n",
    "#         df.rename(columns={'dmrID':'dmr_id','vchr':'chrom','vstart':'position','pval':'pvalues'},inplace=True)\n",
    "#         df['snp']=df.apply(lambda x:tuple([x.chrom,x.position]),axis=1).map(vpos2snp)\n",
    "#         df['MAF']=df.snp.map(maf_dict)\n",
    "#         df=df.loc[df.snp.notna()]\n",
    "#         if not os.path.exists(outfile):\n",
    "#             df.to_csv(outfile,sep='\\t',index=False)\n",
    "#         else:\n",
    "#             df.to_csv(outfile,sep='\\t',index=False,header=False,mode='a')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6e09d49-40a4-49f2-a24e-21d321b0d568",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## Preprocessing UCSC GWAS (Skipped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "48a6e227-3e49-4c6a-a8d9-0ce69426b556",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clinvarCnv.bb\t gdcCancer.bb\tgwasCatalog.sql\n",
      "clinvarMain.bb\t gdcCancer.bed\tgwasCatalog.trackDb.tab.gz\n",
      "clinvarMain.bed  genomeinfo.\tgwasCatalog.txt.gz\n"
     ]
    }
   ],
   "source": [
    "! ls /gale/netapp/DoD/wenliang/database/ref/hg38/ucsc/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7aaf6f19-7c25-4e28-9614-599c7e4e4834",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23\n",
      "bin,chrom,chromStart,chromEnd,name,pubMedID,author,pubDate,journal,title,trait,initSample,replSample,region,genes,riskAllele,riskAlFreq,pValue,pValueDesc,orOrBeta,ci95,platform,cnv\n"
     ]
    }
   ],
   "source": [
    "with open(\"/gale/netapp/DoD/wenliang/database/ref/hg38/ucsc/gwasCatalog.sql\",'r') as f:\n",
    "    lines=f.readlines()\n",
    "cols=[]\n",
    "for line in lines[1:-3]:\n",
    "    cols.append(line.lstrip().split(' ')[0].replace('`',''))\n",
    "print(len(cols))\n",
    "print(','.join(cols))\n",
    "usecols=['chrom','chromStart','chromEnd','name','trait','genes','initSample','replSample','riskAllele','riskAlFreq','pValue','orOrBeta','cnv']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b99c0298-0258-4e25-87c7-4b10fd3a1ebe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>trait</th>\n",
       "      <th>genes</th>\n",
       "      <th>initSample</th>\n",
       "      <th>replSample</th>\n",
       "      <th>riskAllele</th>\n",
       "      <th>riskAlFreq</th>\n",
       "      <th>pValue</th>\n",
       "      <th>orOrBeta</th>\n",
       "      <th>cnv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:845016-845017</td>\n",
       "      <td>chr1</td>\n",
       "      <td>845016</td>\n",
       "      <td>845017</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>Morning vs. evening chronotype</td>\n",
       "      <td>LINC01128</td>\n",
       "      <td>8,724 European ancestry evening chronotype ind...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs141175086-C</td>\n",
       "      <td>0.998</td>\n",
       "      <td>4.000000e-08</td>\n",
       "      <td>2.160000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:946652-946653</td>\n",
       "      <td>chr1</td>\n",
       "      <td>946652</td>\n",
       "      <td>946653</td>\n",
       "      <td>rs2272756</td>\n",
       "      <td>Skin reflectance (Melanin index)</td>\n",
       "      <td>NR</td>\n",
       "      <td>720 South Asian ancestry individuals</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2272756-A</td>\n",
       "      <td>NR</td>\n",
       "      <td>3.000000e-06</td>\n",
       "      <td>0.324533</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:959138-959139</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959138</td>\n",
       "      <td>959139</td>\n",
       "      <td>rs115438739</td>\n",
       "      <td>California verbal learning test score</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>523 European ancestry schizophrenia cases, 100...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs115438739-A</td>\n",
       "      <td>NR</td>\n",
       "      <td>7.000000e-06</td>\n",
       "      <td>7.540324</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:959192-959193</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959192</td>\n",
       "      <td>959193</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>Pancreatic cancer</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>9,040 European ancestry cases, 12,496 European...</td>\n",
       "      <td>up to 2,737 cases, up to 4,752 controls</td>\n",
       "      <td>rs13303010-G</td>\n",
       "      <td>0.11</td>\n",
       "      <td>8.000000e-14</td>\n",
       "      <td>1.260000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:960325-960326</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960325</td>\n",
       "      <td>960326</td>\n",
       "      <td>rs13303327</td>\n",
       "      <td>Heel bone mineral density</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>426,824 British ancestry individuals</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs13303327-G</td>\n",
       "      <td>0.0857658</td>\n",
       "      <td>4.000000e-13</td>\n",
       "      <td>0.023797</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188288</th>\n",
       "      <td>chrY:56944004-56944005</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56944004</td>\n",
       "      <td>56944005</td>\n",
       "      <td>rs306890</td>\n",
       "      <td>Triglycerides</td>\n",
       "      <td>NR</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs306890-T</td>\n",
       "      <td>0.754</td>\n",
       "      <td>8.000000e-12</td>\n",
       "      <td>0.034000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188289</th>\n",
       "      <td>chrY:56944004-56944005</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56944004</td>\n",
       "      <td>56944005</td>\n",
       "      <td>rs306890</td>\n",
       "      <td>Total cholesterol levels</td>\n",
       "      <td>NR</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs306890-T</td>\n",
       "      <td>NR</td>\n",
       "      <td>1.000000e-22</td>\n",
       "      <td>0.048000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188290</th>\n",
       "      <td>chrY:56944004-56944005</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56944004</td>\n",
       "      <td>56944005</td>\n",
       "      <td>rs306890</td>\n",
       "      <td>Total cholesterol levels</td>\n",
       "      <td>NR</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs306890-T</td>\n",
       "      <td>0.754</td>\n",
       "      <td>7.000000e-19</td>\n",
       "      <td>0.049000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188291</th>\n",
       "      <td>chrY:56944004-56944005</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56944004</td>\n",
       "      <td>56944005</td>\n",
       "      <td>rs306890</td>\n",
       "      <td>Low density lipoprotein cholesterol levels</td>\n",
       "      <td>NR</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs306890-T</td>\n",
       "      <td>NR</td>\n",
       "      <td>4.000000e-19</td>\n",
       "      <td>0.045000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188292</th>\n",
       "      <td>chrY:56944004-56944005</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56944004</td>\n",
       "      <td>56944005</td>\n",
       "      <td>rs306890</td>\n",
       "      <td>Low density lipoprotein cholesterol levels</td>\n",
       "      <td>NR</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs306890-T</td>\n",
       "      <td>0.754</td>\n",
       "      <td>9.000000e-15</td>\n",
       "      <td>0.044000</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>188293 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        var_id chrom  chromStart  chromEnd         name  \\\n",
       "0           chr1:845016-845017  chr1      845016    845017  rs141175086   \n",
       "1           chr1:946652-946653  chr1      946652    946653    rs2272756   \n",
       "2           chr1:959138-959139  chr1      959138    959139  rs115438739   \n",
       "3           chr1:959192-959193  chr1      959192    959193   rs13303010   \n",
       "4           chr1:960325-960326  chr1      960325    960326   rs13303327   \n",
       "...                        ...   ...         ...       ...          ...   \n",
       "188288  chrY:56944004-56944005  chrY    56944004  56944005     rs306890   \n",
       "188289  chrY:56944004-56944005  chrY    56944004  56944005     rs306890   \n",
       "188290  chrY:56944004-56944005  chrY    56944004  56944005     rs306890   \n",
       "188291  chrY:56944004-56944005  chrY    56944004  56944005     rs306890   \n",
       "188292  chrY:56944004-56944005  chrY    56944004  56944005     rs306890   \n",
       "\n",
       "                                             trait      genes  \\\n",
       "0                   Morning vs. evening chronotype  LINC01128   \n",
       "1                 Skin reflectance (Melanin index)         NR   \n",
       "2            California verbal learning test score      NOC2L   \n",
       "3                                Pancreatic cancer      NOC2L   \n",
       "4                        Heel bone mineral density     KLHL17   \n",
       "...                                            ...        ...   \n",
       "188288                               Triglycerides         NR   \n",
       "188289                    Total cholesterol levels         NR   \n",
       "188290                    Total cholesterol levels         NR   \n",
       "188291  Low density lipoprotein cholesterol levels         NR   \n",
       "188292  Low density lipoprotein cholesterol levels         NR   \n",
       "\n",
       "                                               initSample  \\\n",
       "0       8,724 European ancestry evening chronotype ind...   \n",
       "1                    720 South Asian ancestry individuals   \n",
       "2       523 European ancestry schizophrenia cases, 100...   \n",
       "3       9,040 European ancestry cases, 12,496 European...   \n",
       "4                    426,824 British ancestry individuals   \n",
       "...                                                   ...   \n",
       "188288  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "188289  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "188290  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "188291  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "188292  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "\n",
       "                                     replSample     riskAllele riskAlFreq  \\\n",
       "0                                           NaN  rs141175086-C      0.998   \n",
       "1                                           NaN    rs2272756-A         NR   \n",
       "2                                           NaN  rs115438739-A         NR   \n",
       "3       up to 2,737 cases, up to 4,752 controls   rs13303010-G       0.11   \n",
       "4                                           NaN   rs13303327-G  0.0857658   \n",
       "...                                         ...            ...        ...   \n",
       "188288                                      NaN     rs306890-T      0.754   \n",
       "188289                                      NaN     rs306890-T         NR   \n",
       "188290                                      NaN     rs306890-T      0.754   \n",
       "188291                                      NaN     rs306890-T         NR   \n",
       "188292                                      NaN     rs306890-T      0.754   \n",
       "\n",
       "              pValue  orOrBeta cnv  \n",
       "0       4.000000e-08  2.160000   N  \n",
       "1       3.000000e-06  0.324533   N  \n",
       "2       7.000000e-06  7.540324   N  \n",
       "3       8.000000e-14  1.260000   N  \n",
       "4       4.000000e-13  0.023797   N  \n",
       "...              ...       ...  ..  \n",
       "188288  8.000000e-12  0.034000   N  \n",
       "188289  1.000000e-22  0.048000   N  \n",
       "188290  7.000000e-19  0.049000   N  \n",
       "188291  4.000000e-19  0.045000   N  \n",
       "188292  9.000000e-15  0.044000   N  \n",
       "\n",
       "[188293 rows x 14 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv(\"/gale/netapp/DoD/wenliang/database/ref/hg38/ucsc/gwasCatalog.txt.gz\",sep='\\t',header=None,names=cols)\n",
    "df=df.loc[:,usecols]\n",
    "df.insert(0,'var_id',df.chrom.map(str)+':'+df.chromStart.map(str)+'-'+df.chromEnd.map(str))\n",
    "df.to_csv(\"GWAS.tsv\",sep='\\t',index=False)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "f337e002-b9ac-4b65-8e97-6f2e56ae21fd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4151\n"
     ]
    }
   ],
   "source": [
    "print(df.trait.nunique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1df32355-cc0a-434d-9216-b615c0e3fe9b",
   "metadata": {},
   "source": [
    "## Prepare Input of coloc for cis-mQTL & GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8200ec7d-f36e-4525-9697-aa198caf91fa",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "m3c",
   "language": "python",
   "name": "m3c"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
